libs/httpsignature/encapsulate_test.go

```diff
@@ -0,0 +1,119 @@
+package httpsignature
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/hex"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEncapsulateRequest(t *testing.T) {
+	var pubKey Ed25519PubKey
+	pubKey, err := hex.DecodeString("e7876fd5cc3a228dad634816f4ec4b80a258b2a552467e5d26f30003211bc45d")
+	assert.NoError(t, err)
+
+	sig := `keyId="primary",algorithm="ed25519",headers="digest foo",signature="HvrmTu+A96H46IPZAYC2rmqRSgmgUgCcyPcnCikX0eGPSC6Va5jyr3blRLjpbGk6UMJ1FXckdWFnJxkt36gkBA=="`
+	digest := "SHA-256=RK/0qy18MlBSVnWgjwz6lZEWjP/lF5HF9bvEF8FabDg="
+	body := []byte("{\"hello\": \"world\"}\n")
+
+	r, err := http.NewRequest("GET", "http://example.org/foo", ioutil.NopCloser(bytes.NewBuffer(body)))
+	assert.NoError(t, err)
+
+	r.Header.Set("Foo", "bar")
+	r.Header.Set("Digest", digest)
+	r.Header.Set("Signature", sig)
+
+	// Header that does not need to be passed
+	r.Header.Set("Bar", "foo")
+
+	er, err := EncapsulateRequest(r)
+	assert.NoError(t, err)
+
+	expectedHeaders := map[string]string{
+		"digest":    digest,
+		"signature": sig,
+		"foo":       "bar",
```

Comment:
is this normalizing `Foo` to `foo` in an intentional way? It looks like we're setting the header to `Foo` on line 27 then expecting lowercase here.

Not sure if this is expected behavior of HTTP headers so just want to check.

Reply:
the keys actually end up coming from the headers specified in the httpsignature which per the spec: 
>    If specified, it SHOULD be a lowercased, quoted list of HTTP header fields, separated by a single space character.

it's a bit odd but since we use Golang header canonicalization when encapsulating and extracting the exact capitalization doesn't end up mattering